.github/workflows/ci-rest.yml

```diff
@@ -0,0 +1,50 @@
+name: REST CI
+
+on:
+  workflow_call:
+  workflow_dispatch:
+    inputs:
+      ref:
+        type: string
+        required: false
+        description: TileDB core reference to test against REST CI.
+  push:
+    branches:
+      - main
+      - release-*
```

Comment:
The other jobs under `full-ci.yaml` just use 
```
on:
  workflow_call:
```
Is there a reason we need `workflow_dispatch` and `push` sections here?